packages/labs/ssr/src/lib/render-result-readable-stream.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {RenderResult} from './render-result.js';
+
+type RenderResultIterator = Iterator<string | Promise<RenderResult>>;
+
+/**
+ * A Readable that reads from a RenderResult.
+ */
+export const renderResultReadableStream = (result: RenderResult) => {
```

Comment:
In https://github.com/lit/lit/blob/main/packages/labs/ssr/src/lib/render-result-readable.ts we use subclassing so that you construct a new `RenderResultReadable`. I think we should make these modules consistent*. So should we use subclassing here? Or should we add a factory function to `render-result-readable.ts`?

*overriding consistency would be being idiomatic - so if users of the different stream APIs have different expectations, we should meet those.

Reply:
I have a issues with types when try to do subclassing, some refs:

https://github.com/microsoft/TypeScript/issues/39051
https://github.com/whatwg/streams/issues/778

I can try to organize it in a different way, if there are ideas 🫠